lib/services/publish-service.ts

```diff
@@ -0,0 +1,208 @@
+        }).future<void>()();
+    }
+
+    public removeConnection(idOrName: string): IFuture<void> {
+        return (() => {
+            var id = parseInt(idOrName) - 1;
+            var message : string;
+            if (!isNaN(id)) {
+                this.validateId(id);
+
+                var connectionIndex = _.findIndex(this.allPublishConnections, this.ftpPublishConnections[id]);
+                this.allPublishConnections.splice(connectionIndex, 1);
+                message = util.format("Connection '%s' successfully removed.", this.ftpPublishConnections[id].name);
+            } else {
+                var name = idOrName;
+                var connectionsToBeRemoved = _.filter(this.ftpPublishConnections, c => c.name === name);
+
+                if (connectionsToBeRemoved.length > 1) {
+                    var connectionNameUrls = _.map(connectionsToBeRemoved, c => c.name + '(' + c.publishUrl + ')');
+                    connectionNameUrls.splice(connectionNameUrls.length, 0, PublishService.REMOVE_ALL_CONNECTIONS_MESSAGE);
+
+                    var chosenConnectionNameUrl = this.$prompter.promptForChoice("Which connection would you like to remove?", connectionNameUrls).wait();
+                    if (chosenConnectionNameUrl === PublishService.REMOVE_ALL_CONNECTIONS_MESSAGE) {
+                        _.remove(this.allPublishConnections, c => c.name === name && c.type === PublishService.ALLOWED_CONNECTION_TYPE);
+                        message = util.format("All connections named '%s' successfully removed.", name);
+                    } else {
+                        var chosenConnectionIndex = _.findIndex(this.allPublishConnections, c => c.name + '(' + c.publishUrl + ')' === chosenConnectionNameUrl);
+                        this.allPublishConnections.splice(chosenConnectionIndex, 1);
+                        message = util.format("Connection '%s' successfully removed.", chosenConnectionNameUrl);
+                    }
+                } else if (connectionsToBeRemoved.length === 1) {
+                    _.remove(this.allPublishConnections, connectionsToBeRemoved[0]);
+                    message = util.format("Connection '%s' successfully removed.", connectionsToBeRemoved[0].name);
+                } else {
+                    this.$errors.failWithoutHelp("No connections named '%s' found.", name);
+                }
+            }
+
+            this.savePublishJsonFile().wait();
+            this.$logger.info(message);
+        }).future<void>()();
+    }
+
+    private readConnections(): IFuture<void> {
+        return (() => {
+            var projectDir = this.$project.getProjectDir().wait();
+            var publishFilePath = path.join(projectDir, PublishService.JSON_PUBLISH_FILE_NAME);
+            try {
+                this.allPublishConnections = this.$fs.readJson(publishFilePath).wait();
+            } catch (err) {
+                if (err.code === 'ENOENT') {
```

Comment:
I disagree, a file system entity may exist but is unusable. Typical examples are - a directory with the same name, file is being locked (usually a bad anti-virus, but text editors are also known to do this), there may be attributes or ACLs which deny reading the file. One must write the try / catch anyway so the exists check is unneeded.